From e894e1675e5c59c125614fb89ebf17c64d8f4c48 Mon Sep 17 00:00:00 2001
From: Anuj Mittal <anujx.mittal@intel.com>
Date: Fri, 1 Apr 2016 11:34:19 +0800
Subject: [PATCH] mraa: point to correct install paths

Use PYTHON_SITEPACKAGES_DIR variable from python-dir bbclass for
PYTHON_SITE_DIR and use generic LIB_INSTALL_DIR path.

Signed-off-by: Anuj Mittal <anujx.mittal@intel.com>
---
 src/javascript/CMakeLists.txt | 4 ++--
 src/python/CMakeLists.txt     | 6 +-----
 2 files changed, 3 insertions(+), 7 deletions(-)

diff --git a/src/javascript/CMakeLists.txt b/src/javascript/CMakeLists.txt
index d54d77a..09714ef 100644
--- a/src/javascript/CMakeLists.txt
+++ b/src/javascript/CMakeLists.txt
@@ -57,11 +57,11 @@ endif ()
 # If a CMAKE_INSTALL_PREFIX has NOT been provided, set NODE_MODULE_INSTALL_PATH
 # base on the NODE_ROOT_DIR.
 if (CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
-  set (NODE_MODULE_INSTALL_PATH ${NODE_ROOT_DIR}/lib/node_modules/mraa/)
+  set (NODE_MODULE_INSTALL_PATH ${NODE_ROOT_DIR}/${BASE_LIB_INSTALL_DIR}/node_modules/mraa/)
 # If a CMAKE_INSTALL_PREFIX has been provided, set NODE_MODULE_INSTALL_PATH
 # relative to the provide install directory.
 else (CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
-  set (NODE_MODULE_INSTALL_PATH ${CMAKE_INSTALL_PREFIX}/lib/node_modules/mraa/)
+  set (NODE_MODULE_INSTALL_PATH ${CMAKE_INSTALL_PREFIX}/${BASE_LIB_INSTALL_DIR}/node_modules/mraa/)
 endif (CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
 message (STATUS "INFO - install NODE modules to ${NODE_MODULE_INSTALL_PATH}")
 
diff --git a/src/python/CMakeLists.txt b/src/python/CMakeLists.txt
index c51ccd1..6db54f6 100644
--- a/src/python/CMakeLists.txt
+++ b/src/python/CMakeLists.txt
@@ -34,11 +34,7 @@ execute_process (
    OUTPUT_VARIABLE PYTHON_PREFIX
 )
 file (TO_CMAKE_PATH "${PYTHON_PREFIX}" PYTHON_PREFIX)
-execute_process (
-   COMMAND ${PYTHON_EXECUTABLE} -c
-       "import site, sys; sys.stdout.write(site.getusersitepackages().replace(site.getuserbase(), site.PREFIXES[-1]))"
-   OUTPUT_VARIABLE PYTHON_SITE_DIR
-)
+
 file (TO_CMAKE_PATH "${PYTHON_SITE_DIR}" PYTHON_SITE_DIR)
 string (REGEX REPLACE "^${PYTHON_PREFIX}/" ""
    PYTHON_SITE_DIR "${PYTHON_SITE_DIR}")
-- 
1.9.1

